tests/advanced_visibility_test.go

```diff
@@ -540,59 +543,99 @@ func (s *advancedVisibilitySuite) TestListWorkflow_KeywordQuery() {
 	s.NoError(err)
 	s.Len(resp.GetExecutions(), 0)
 
-	if s.isElasticsearchEnabled {
-		// LIKE is supported on Elasticsearch only.
+	// Prefix search
+	listRequest = &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: s.namespace,
+		PageSize:  defaultPageSize,
+		Query:     `CustomKeywordField STARTS_WITH "justice"`,
+	}
+	resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	s.NoError(err)
+	s.Len(resp.GetExecutions(), 1)
+	s.Equal(id, resp.Executions[0].GetExecution().GetWorkflowId())
+	s.Equal(wt, resp.Executions[0].GetType().GetName())
+	s.Equal(searchAttr, resp.Executions[0].GetSearchAttributes())
 
-		// LIKE exact match on Keyword (supported)
-		listRequest = &workflowservice.ListWorkflowExecutionsRequest{
-			Namespace: s.namespace,
-			PageSize:  defaultPageSize,
-			Query:     `CustomKeywordField LIKE "%justice for all%"`,
-		}
-		resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
-		s.NoError(err)
-		s.Len(resp.GetExecutions(), 1)
+	listRequest = &workflowservice.ListWorkflowExecutionsRequest{
+		Namespace: s.namespace,
+		PageSize:  defaultPageSize,
+		Query:     `CustomKeywordField NOT STARTS_WITH "justice"`,
+	}
+	resp, err = s.engine.ListWorkflowExecutions(NewContext(), listRequest)
+	s.NoError(err)
+	s.Len(resp.GetExecutions(), 0)
```

Comment:
@tdeebswihart These are functional tests.

Reply:
These are the ones I raise an issue about earlier. These tests look for the presence of _a_ record (or no records) but do not verify that the returned record is what we expect.

Can you update these to ensure the returned execution actually matches the filter?I've seen tests like this (that look for the presence of _a_ record) hide bugs where the incorrect record was returned

Reply:
I already added checks. See L555-557.

Reply:
D'oh, thank you